packages/frontend/src/components/common/Table/getGroupedRowModelLightdash.ts

```diff
@@ -0,0 +1,225 @@
+import { createRow } from '@tanstack/table-core/src/core/row';
+import {
+    type Row,
+    type RowData,
+    type RowModel,
+    type Table,
+} from '@tanstack/table-core/src/types';
+import { flattenBy, memo } from '@tanstack/table-core/src/utils';
+
+// Customizes row grouping based on a clone of
+// @tanstack/table-core/src/utils/getGroupedRowModel.ts
+// Per TanStack's recommendation:
+// https://tanstack.com/table/v8/docs/guide/row-models#customizefork-row-models
+
+function groupBy<TData extends RowData>(rows: Row<TData>[], columnId: string) {
+    const groupMap = new Map<any, Row<TData>[]>();
+
+    return rows.reduce((map, row) => {
+        // This line changed from original since getGroupingValue does not exist here as it should.
+        // const resKey = `${row.getGroupingValue(columnId)}`
+        const resKey = (row.original as any)[columnId].value.raw;
+
+        const previous = map.get(resKey);
+        if (!previous) {
+            map.set(resKey, [row]);
+        } else {
+            previous.push(row);
+        }
+        return map;
+    }, groupMap);
+}
+
+export function getGroupedRowModelLightdash<TData extends RowData>(): (
+    table: Table<TData>,
+) => () => RowModel<TData> {
+    return (table) =>
+        memo(
+            () => [table.getState().grouping, table.getPreGroupedRowModel()],
+            (grouping, rowModel) => {
+                if (!rowModel.rows.length || !grouping.length) {
+                    return rowModel;
+                }
+
+                // Filter the grouping list down to columns that exist
+                const existingGrouping = grouping.filter((columnId) =>
+                    table.getColumn(columnId),
+                );
+
+                const groupedFlatRows: Row<TData>[] = [];
+                const groupedRowsById: Record<string, Row<TData>> = {};
+                // const onlyGroupedFlatRows: Row[] = [];
+                // const onlyGroupedRowsById: Record<RowId, Row> = {};
+                // const nonGroupedFlatRows: Row[] = [];
+                // const nonGroupedRowsById: Record<RowId, Row> = {};
```

Comment:
can we remove commented out code from this file? or maybe we could add a comment for the context why it's commented out?

Reply:
This file is a copy of https://github.com/TanStack/table/blob/main/packages/table-core/src/utils/getGroupedRowModel.ts which has these lines commented out. We can remove the lines though. The only reason to keep them is to make comparison to the source file more straightforward in the future.

Reply:
fair enough, let's keep them! thanks Tony!

Reply:
Actually, I think it is no big deal to remove them. It does not make the comparison to the original especially difficult.